### ISSUE TITLE
[NBug] Index was out of range. Must be non-negative and less than ...

### ISSUE THREAD

**Issue by PavelA85** [gitextensions/gitextensions#7374]
I was clicking around. Some commits were selected, while I tried to `fetch all` and got this exception
## Error Details
```
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.Collections.ArrayList.get_Item(Int32 index)
   at System.Windows.Forms.DataGridViewRowCollection.SharedRow(Int32 rowIndex)
   at System.Windows.Forms.DataGridViewRowCollection.get_Item(Int32 index)
   at GitUI.RevisionGridControl.OnGridViewCellMouseDown(Object sender, DataGridViewCellMouseEventArgs e)
   at System.Windows.Forms.DataGridView.OnCellMouseDown(DataGridViewCellMouseEventArgs e)
   at System.Windows.Forms.DataGridView.OnMouseDown(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseDown(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

## Additional information
just clicking around

## Environment
- Git Extensions 3.2.1.6628
- Build bc07663c3c893f1d896520e0bdcfbe35cfe4b729
- Git 2.23.0.windows.1
- Microsoft Windows NT 10.0.16299.0
- .NET Framework 4.8.4018.0
- DPI 96dpi (no scaling)

**Comment by gerhardol** [comment]
Is ```_latestSelectedRowIndex``` changed in OnGridViewSelectionChanged while OnGridViewCellMouseDown runs?

**Comment by NN---** [comment]
Still happens with 3.3.1
```
- Git Extensions 3.3.1.7897
- Build 5a97671645532bcedc443bac7b727f40db47cb5c
- Git 2.26.0.windows.1
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4150.0
- DPI 96dpi (no scaling)
```

**Comment by RussKie** [comment]
Steps to reproduce?

**Comment by NN---** [comment]
Unfortunately I don't have good steps, it just happened right now without doing anything special

**Comment by pmiossec** [comment]
I don't understand why it happens but looking at the code:

```csharp
            if (_latestSelectedRowIndex == hti.RowIndex && _gridView.Rows[_latestSelectedRowIndex].Selected)
            {
                return;
            }
```

I think that we should protect the access with index to the `Rows`

**Comment by gerhardol** [comment]
As other issues point back to this:
There was a correction for #7374 in #7921 that is in 3.4.
Still a few occurrences in 3.4
Corrected in #8534, included since 3.5 RC1
